### PR TITLE
Add `--style arrows` drawing style

### LINKED ISSUE #3: Use Colored axis instead to illustrate the rotation
The illustration example in the readme is very impressive. However, I feel the x, y, z ring is a little bit overwhelming to me. 
I am NOT suggesting not to showing the ring. But I suggest can we have another mode that only showing the rotation effect on the axis, i.e. like show original +x, +y, +z axes in bold blue, red, green only.

I think colored axis can be useful. For example, for all 24 single qubit Clifford gates, they just permute the axis. Showing the axis only can be very clean and informative. Related to https://github.com/quantumlib/Cirq/pull/4050 that connecting the clifford gate to phased xz gate. Also related to visualize the `decompose_rotation` that you wrote there.

### PR DESCRIPTION
**Pull request by BichengYing**
Fix #3  Usage example: `animate_bloch example x --style arrows`

### CODE REVIEW

File: bloch_sphere/animate_bloch.py [hunk 1]
```diff
@@ -215,7 +215,8 @@ def draw_frame(*args, background='white', idPrefix='d', w=624, h=None,
 def draw_bloch_sphere(d, inner_proj=euclid3d.identity(3), label='', axis=None,
                       rot_proj=None, rot_deg=180,
                       outer_labels=(), inner_labels=(),
-                      extra_opacity=1, inner_opacity=1, background='white'):
+                      extra_opacity=1, inner_opacity=1, background='white',
+                      style=None):
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py]: Later, `None` will break the `.get` check later.  Add a line here `if style is None: style = {}`.

**Review by cduck — COMMENTED** [review]
Suggestion for the new command line argument: Instead of the `--arrow-style` flag, use something like `--style arrows` so new styles can be added in the future.  Another option is `--inner-style arrows --outer-style sphere` (both of these would default to `sphere`).

```python
parser.add_argument('style', type=str, choices=['sphere', 'arrows'], default='sphere', help='...')
```

**Comment by BichengYing** [comment]
Agree, `--style arrows` sounds better than boolean `--arrow-style`. PTAL.

File: bloch_sphere/animate_bloch.py (hunk 1 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: No longer needed after `--style` change

File: bloch_sphere/animate_bloch.py [hunk 2]
```diff
@@ -419,8 +445,12 @@ def run_from_command_line():
         'Save an mp4 video instead of a GIF')
     parser.add_argument('--fps', type=float, default=20, help=
         'Sets the animation frame rate')
+    parser.add_argument('--style', type=str, choices=['sphere', 'arrows'], 
+        default='sphere', help='Different style of showing rotated bloch sphere. '
+        'Supported options are [sphere, arrows].')
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py]: `bloch_sphere --help` should automatically generate a description of the available choices.
```suggestion
```

File: bloch_sphere/animate_bloch.py [hunk 3]
```diff
@@ -419,8 +445,12 @@ def run_from_command_line():
         'Save an mp4 video instead of a GIF')
     parser.add_argument('--fps', type=float, default=20, help=
         'Sets the animation frame rate')
+    parser.add_argument('--style', type=str, choices=['sphere', 'arrows'], 
+        default='sphere', help='Different style of showing rotated bloch sphere. '
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py]: Suggested help: `'The style to draw the Bloch sphere.  E.g. draw the whole sphere or just draw the axis arrows.'`

File: bloch_sphere/animate_bloch.py [hunk 4]
```diff
@@ -419,8 +445,12 @@ def run_from_command_line():
         'Save an mp4 video instead of a GIF')
     parser.add_argument('--fps', type=float, default=20, help=
         'Sets the animation frame rate')
+    parser.add_argument('--style', type=str, choices=['sphere', 'arrows'],
```
Comment (cduck) [inline bloch_sphere/animate_bloch.py]: Nit: Trailing space.

File: bloch_sphere/animate_bloch_compare.py [hunk 5]
```diff
@@ -132,11 +132,14 @@ def run_from_command_line():
         'Save an mp4 video instead of a GIF')
     parser.add_argument('--fps', type=float, default=20, help=
         'Sets the animation frame rate')
+    parser.add_argument('--style', type=str, choices=['sphere', 'arrows'], 
+        default='sphere', help='Different style of showing rotated bloch sphere. '
+        'Supported options are [sphere, arrows].')
```
Comment (cduck) [inline bloch_sphere/animate_bloch_compare.py]: Same as the above three comments.

**Review by cduck — CHANGES_REQUESTED** [review]
Looks good.  Just a few minor things.

Do you want to add an example using this style to the end of **Usage** in the readme with a GIF output?  (Can be a separate PR.  Otherwise I'll add it when I have time.)

File: bloch_sphere/animate_bloch_compare.py (hunk 5 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch_compare.py]: Done.

File: bloch_sphere/animate_bloch.py (hunk 4 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: Done.

File: bloch_sphere/animate_bloch.py (hunk 3 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: Done.

File: bloch_sphere/animate_bloch.py (hunk 2 above)
Comment (BichengYing) [reply bloch_sphere/animate_bloch.py]: Updated according to later comments

**Comment by BichengYing** [comment]
No problem. It is easy to update README. What kind of gif do you want to put in the Readme? Give me the command. 
I think we can include the commands like `Rx;0.5` there to illustrate it.

**Comment by cduck** [comment]
I found a bug while testing this.  I hope you don't mind that I fixed it and added the example.  Let me know what you think or if you want to make any other changes before I merge.

**Comment by BichengYing** [comment]
Thanks for fixing the bug. Just one minor comments in README, you have line
`animate_bloch ~/Downloads/rx_gate_arrows --style arrows ry,0.666667 ry,0.666667 ry,0.666667`.
Do you want to keep `~/Downloads/`? Also, the name should be `ry_gate_arrows` instead of `rx`?

**Comment by cduck** [comment]
Good catch.  I need to review my own code.

**Comment by cduck** [comment]
Published in [version 0.4.0](https://pypi.org/project/bloch-sphere/0.4.0/).

**Pull request merged**